packages/two_dimensional_scrollables/lib/src/tree_view/tree.dart

```diff
@@ -0,0 +1,1104 @@
+  /// collapsing node animation.
+  static const Curve defaultAnimationCurve = Curves.linear;
+
+  /// A default [Duration] of 150 milliseconds, which is used in the tree's
+  /// expanding and collapsing node animation.
+  static const Duration defaultAnimationDuration = Duration(milliseconds: 150);
+
+  /// A wrapper method for triggering the expansion or collapse of a
+  /// [TreeViewNode].
+  ///
+  /// Use as part of [TreeView.defaultTreeNodeBuilder] to wrap the leading icon
+  /// of parent [TreeViewNode]s such that tapping on it triggers the animation.
+  ///
+  /// If defining your own [TreeView.treeNodeBuilder], this method can be used
+  /// to wrap any part, or all, of the returned widget in order to trigger the
+  /// change in state for the node when tapped.
+  ///
+  /// The gesture uses [HitTestBehavior.translucent], so as to not conflict
+  /// with any [TreeRow.recognizerFactories] or other interactive content in the
+  /// [TreeRow].
+  static Widget wrapChildToToggleNode({
+    required TreeViewNode<Object?> node,
+    required Widget child,
+  }) {
+    return Builder(builder: (BuildContext context) {
+      return GestureDetector(
+        behavior: HitTestBehavior.translucent,
+        onTap: () {
+          TreeViewController.of(context).toggleNode(node);
+        },
+        child: child,
+      );
+    });
+  }
+
+  /// Returns the fixed height, default [TreeRow] for rows in the tree,
+  /// which is 40 pixels.
+  ///
+  /// Used by [TreeView.treeRowBuilder].
+  static TreeRow defaultTreeRowBuilder(TreeViewNode<Object?> node) {
+    return const TreeRow(
+      extent: FixedTreeRowExtent(_kDefaultRowExtent),
+    );
+  }
+
+  /// Returns the default tree row for a given [TreeViewNode].
+  ///
+  /// Used by [TreeView.treeNodeBuilder].
+  ///
+  /// This will return a [Row] containing the [toString] of
+  /// [TreeViewNode.content]. If the [TreeViewNode] is a parent of additional
+  /// nodes, an arrow icon will precede the content, and will trigger an expand
+  /// and collapse animation when tapped based on the
+  /// [TreeView.toggleAnimationStyle].
+  static Widget defaultTreeNodeBuilder(
+    BuildContext context,
+    TreeViewNode<Object?> node,
+    AnimationStyle toggleAnimationStyle,
+  ) {
+    final Duration animationDuration =
+        toggleAnimationStyle.duration ?? TreeView.defaultAnimationDuration;
+    final Curve animationCurve =
+        toggleAnimationStyle.curve ?? TreeView.defaultAnimationCurve;
+    final int index = TreeViewController.of(context).getActiveIndexFor(node)!;
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Row(children: <Widget>[
+        // Icon for parent nodes
+        TreeView.wrapChildToToggleNode(
+          node: node,
+          child: SizedBox.square(
+            dimension: 30.0,
+            child: node.children.isNotEmpty
+                ? AnimatedRotation(
+                    key: Key('$index - ${node.content}'),
```

Comment:
Just index will not work since the index can change at anytime when a node is toggled open or closed.

Reply:
The current key will also change whenever the index changes. What does including the content help with?

Reply:
Yeah, I thought this could cause a conflict where one with an index of 3 is animating, then the indexes change and the new keyed-3 node is mismatched and creates an issue with the animation. I haven't been able to reproduce this though, so I updated it to just the index.